packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -152,17 +153,35 @@ const getContextMenuStyles = stylesFactory((theme: GrafanaTheme) => {
 export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClose, items, renderHeader }) => {
   const theme = useContext(ThemeContext);
   const menuRef = useRef(null);
+  const [positionStyles, setPositionStyles] = useState(null);
+
+  useEffect(() => {
+    if (menuRef) {
+      const rect = menuRef.current.getBoundingClientRect();
+      const OFFSET = 5;
+      const collisions = {
+        right: window.innerWidth < x + rect.width,
+        bottom: window.innerHeight < rect.bottom + rect.height + OFFSET,
+      };
+
+      setPositionStyles({
+        position: 'fixed',
+        left: collisions.right ? x - rect.width - OFFSET : x - OFFSET,
+        top: collisions.bottom ? y - rect.height - OFFSET : y + OFFSET,
+      });
+    }
+  }, [menuRef]);
```

Comment:
I am not sure about this. `menuRef` should be stable and should not change, maybe this should be `menuRef.current`?

Reply:
yeah, you r right